view/rooms/list/ListController.js

```diff
@@ -22,16 +22,25 @@ class ListController extends React.Component {
 
   render() {
     const rooms = this.props.rooms !== undefined ? this.props.rooms : [];
-    rooms.sort((a, b) => {
-      return moment(b.updatedAt) - moment(a.updatedAt);
-    });
     const roomTypes = this.props.roomTypes !== undefined ? this.props.roomTypes : [];
+
+    rooms.sort((a, b) => moment(b.updatedAt) - moment(a.updatedAt));
+
+    if (rooms.length > 0) {
+      return this.renderRooms(rooms, roomTypes);
+    } else {
+      return this.renderRoomTypes(roomTypes);
+    }
+  }
+
+  renderRooms(rooms, roomTypes) {
```

Comment:
Why create a single component when there are 2 different renders?

I think you need to make this component more generic 

like < ListController headers =[one,two,three] rows=[{1,2,3},{1,23}]>
and from the rooms and room types you parse the data and pass it to the component

Reply:
agreed. i'll work on it